### PR TITLE
[CI-2284] Add configuration hash to the xml data

### PR DESCRIPTION
**Pull request by tothszabi**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

The generated xml data representation is not accurate for retried tests. These test runs appear as separate entries in the xml and we need to apply grouping/filtering logic to consume them. 

A simple logic is to simply use the test class name and test name. This works for the regular use cases but there is a new feature which breaks this which is the device configuration tests. This is a handy new Xcode feature where you can tell Xcode to launch the given tests on all language / orientation / interface mode combinations with:
```
override class var runsForEachTargetApplicationUIConfiguration: Bool {
    true
}
```
But tests launched like this will have the same name and identifier and only their configuration will be different. Xcode uses this configuration combo to uniquely identify them. The deploy step was not extracting these and now it will create a checksum from the configuration. 

The backend side grouping can then reliably identify and group related device configuration test runs.

Here is an example of a configuration object extracted from the xcresult file:
```
"configuration" : {
  "_type" : {
    "_name" : "ActionTestConfiguration"
  },
  "values" : {
    "_type" : {
      "_name" : "SortedKeyValueArray"
    },
    "storage" : {
      "_type" : {
        "_name" : "Array"
      },
      "_values" : [
        {
          "_type" : {
            "_name" : "SortedKeyValueArrayPair"
          },
          "key" : {
            "_type" : {
              "_name" : "String"
            },
            "_value" : "XCUIAppearanceMode"
          },
          "value" : {
            "_type" : {
              "_name" : "String"
            },
            "_value" : "1"
          }
        },
        {
          "_type" : {
            "_name" : "SortedKeyValueArrayPair"
          },
          "key" : {
            "_type" : {
              "_name" : "String"
            },
            "_value" : "XCUIDeviceOrientation"
          },
          "value" : {
            "_type" : {
              "_name" : "String"
            },
            "_value" : "1"
          }
        }
      ]
    }
  }
},
```

### Changes

There is a new `ConfigurationHash` field on the xml data struct. This field will be populated by a checksum of the actual configuration taken from the xcresult file.

### CODE REVIEW

File: test/converters/xcresult3/converter.go [hunk 1]
```diff
@@ -200,15 +201,20 @@ func genTestCase(test ActionTestSummaryGroup, xcresultPath, testResultDir string
 		}
 	}
 
+	testSummary, err := test.loadActionTestSummary(xcresultPath)
```
Comment (tothszabi) [inline test/converters/xcresult3/converter.go:204]: Use a `SummaryNotFoundError` and compare it against that.

File: test/testdata/ios_xml_output.golden [hunk 2]
```diff
@@ -1,40 +1,40 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <testsuites>
  <testsuite name="BitriseBasicUITest" tests="4" failures="0" skipped="0" errors="0" time="77.06086790561676">
-  <testcase name="testCollectionView()" classname="BitriseBasicUITest" time="23.776288986206055"></testcase>
-  <testcase name="testInReadTopView()" classname="BitriseBasicUITest" time="34.97628891468048"></testcase>
-  <testcase name="testInterstitialView()" classname="BitriseBasicUITest" time="10.270301938056946"></testcase>
-  <testcase name="testRewardView()" classname="BitriseBasicUITest" time="8.037988066673279"></testcase>
+  <testcase configuration-hash="" name="testCollectionView()" classname="BitriseBasicUITest" time="23.776288986206055"></testcase>
```
Comment (tothszabi) [inline test/testdata/ios_xml_output.golden:4]: Create a test case with the configuration hash.

**Review by lpusok — APPROVED** [review]
Looks ok as far as I can tell.
It seems we reached (already before?) the limits of jUnit format, as far as I can tell configuration-hash is not a standard junit property (https://github.com/testmoapp/junitxml). But it seems enough to tell test cases apart. Do we need the exact orientation etc. displayed? If yes then obviously it will not be enough. For "standard" junit format we could use test properties possibly https://github.com/testmoapp/junitxml#properties-for-suites-and-cases.

Is there an explicit marker to find retried tests (besides the test name)? There could be other edge cases introduced that could break the heuristic to identify retried test.
